tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Reports/BaseReport.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Azure.Sdk.Tools.GitHubIssues.Services.Configuration;
+using Azure.Security.KeyVault.Secrets;
+using GitHubIssues.Helpers;
+using Microsoft.Extensions.Logging;
+using Octokit;
+using Polly;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Sdk.Tools.GitHubIssues.Reports
+{
+    public abstract class BaseReport
+    {
+        protected IConfigurationService ConfigurationService { get; private set; }
+
+        public BaseReport(IConfigurationService configurationService)
+        {
+            ConfigurationService = configurationService;
+        }
+
+        public async Task ExecuteAsync(ILogger log)
+        {
+            Task<IEnumerable<RepositoryConfiguration>> pendingRepositoryConfigurations = ConfigurationService.GetRepositoryConfigurationsAsync();
+            Task<string> pendingGitHubPersonalAccessToken = ConfigurationService.GetGitHubPersonalAccessTokenAsync();
+            Task<string> pendingSendGridToken = ConfigurationService.GetSendGridTokenAsync();
+            Task<string> pendingFromAddress = ConfigurationService.GetFromAddressAsync();
```

Comment:
I would move these inline. Some of those methods might have sync code that will now run at somewhat random times.